src/components/Timeout/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { h, Component } from 'preact'
+
+export type Props = {
+  seconds: number
+  onTimeout: () => void
+}
+
+type State = {
+  hasTimedOut: boolean
+}
+
+export default class Timeout extends Component<Props, State> {
+  private timeoutId?: number
+
+  state = {
+    hasTimedOut: false,
+  }
+
+  componentDidMount(): void {
+    if (!this.timeoutId) {
+      this.clearInactivityTimeout()
+      this.timeoutId = window.setTimeout(
+        () => this.setState({ hasTimedOut: true }),
+        this.props.seconds * 1000
+      )
+    }
+  }
+
+  componentWillUnmount(): void {
+    this.clearInactivityTimeout()
+  }
+
+  componentDidUpdate(_prevProps: Props, prevState: State): void {
```

Comment:
Curious, but why is this given an underscore prefix - to indicate it is unused here?

Reply:
Yes, it's the [default behaviour](https://eslint.org/docs/rules/no-underscore-dangle#allowfunctionparams) of `no-underscore-dangle` ESLint rule